File: DemoMethods/src/main/java/net/ravendb/demo/basic/Edit.java
package net.ravendb.demo.basic;

import net.ravendb.client.IDocumentSession;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.entities.Company;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class Edit {

    @RequestMapping("/Basic/Edit")
    public String edit(@RequestParam (value = "companyId", defaultValue = "1") int companyId) {
        try (IDocumentSession session = DocumentStoreHolder.getStore().openSession()) {

            Company company = session.load(Company.class, companyId);
            company.getAddress().setLine2("Zip 12345");
            session.saveChanges();

            return String.format("Company %s Edited Successfully", companyId);
        }
    }
}

File: DemoServer/Demos/Basic/Edit.cs
using DemoServer.Controllers;
using DemoServer.Entities;
using DemoServer.Helpers;
using Microsoft.AspNetCore.Mvc;

namespace DemoServer.Demos.Basic
{
    public partial class BasicController : BaseController
    {
        [HttpGet]
        [Route("/basic/edit")]
        [Demo("Edit", DemoOutputType.String, demoOrder: 20)]
        public object Edit(string companyId = "companies/1-A")
        {
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                var company = session.Load<Company>(companyId);

                company.Address.Line2 = "Zip 12345";

                session.SaveChanges();

                return $"{companyId} Edited Successfully";
            }
        }

    }
}

